## simulate_pedigree_structure.pl 
This program can be used to simulate pedigrees of random structure
###Arguments are as follows: 
* $ARGV[0] is a '/' delimited emperical distribution of number of kids starting with 0 going up to 13 (ex. 0/5/3/1/0/0/0/0/0/0/0/0/0/0 would be a ratio of 0:5:3:1 of having 0,1,2, or 3 kids)
* $ARGV[1] is the number of generations to simulate
* $ARGV[2] is a '/' delimited emperical distribution of number of partners a person will have kids with starting with 1 and going up to 4 (ex.10/2/1/1 is a ratio of 10:2:1:1 of having 1,2,3, or 4 partners)

###Additional information
* To create larger more complex pedigrees, use 'simulate_pedigree_structure.pl' to simulate pedigrees then combine then using 'merge_lineages.pl'
* Output is written to STDOUT so redirect the output to a file

##merge_lineages.pl
This program is used in conjunction with 'simulate_pedigree_structure.pl' to combine two pedigrees to make a more complex pedigree structures.
###Arguments are as follows:
* $ARGV[0] and $ARGV[1] are pedigree files to be merged to create a larger, more complex pedigree.
###Additional information
* Output is written to STDOUT so redirect the output to a file

## sim_chr_genos.pl
This program can be used to simulate genotypes according to a user defined emperical distribution, an example distribution is included in the file 'frequency_distribution2.csv'
###Arguments are as follows:
* $ARGV[0] is the frequency distribution file
* $ARGV[1] is the number of markers to simulate
* $ARGV[2] is a pedigree file in the format generated by 'simulate_pedigree_structure.pl'
* $ARGV[3] is the probability that a crossover happens between adjacent markers in a given meiosis
###Outputs are as follows:
* 'All_Diploid_genos.csv' are simulated diploid genotypes 
* 'All_Haploid_genos.csv' are the phased haploid genotypes corresponding to the simulated diploid genotypes in 'All_Diploid_genos.csv'
* 'Diploid_phase_genotypes.csv' summarizes the two founding chromosomes each marker in each individual was inherited from that sum to the observed diploid genotype
* 'Haploid_geno_phase.csv' summarizes which founding chromosome eac marker in each individual was inherited from

## frequency_distribution2.csv
This is an example emperical frequency distribution.
This distribution was generated under the infinite-alleles model with neutrality. R script for this is contained in 'Allele_frequency_spectrum_R_script.txt'

## Allele_frequency_spectrum_R_script.txt
This is R script that simulates an emperical frequency distribution under the infinite alleles model with neutrality.

## Summarize_phase.pl
This program summarized the phase of the genotypes generated by 'sim_chr_genos.pl'  
###Additional information
* There are no arguments, but this program requires that you have already run 'sim_chr_genos.pl'
* Output is written to STDOUT so redirect the output to a file

## mask_simulated_genotypes.pl
This program masks genotypes simulated using 'sim_chr_genos.pl'
### Arguments are as follows:
* $ARGV[0] is the pedigree file that correpsonds to the simulated genotypes
###Additional information
* Output is written to STDOUT so redirect the output to a file

## find_paths_2.pl
This program outputs various measures given a pedigree file.
### Arguments are as follows:
* ARGV[0] is the pedigree file
### Outputs are as follows:
* 'All_paths.csv' is a list of every path that a haploid genotype could potentially take through a pedigree
* 'Founder_offspring_counts.csv' lists the number of offspring that could potentially inherit haploid genotypes from each founder
* 'Founder_offspring.csv' lists the offspring that could potentially inherit haploid genotypes from each founder
* 'Offspring_founders.csv' lists the founders that each pedigree member could potentially inherit haploid genotypes from

## Count_kids.pl
This program outputs the number of kids each individual in the pedigree has. The motivation for this program is so that you can estimate an emperical distribution of the number of kids for 'simulate_pedigree_structure.pl from real existing pedigrees
### Arguments are as follows:
* ARGV[0] is pedigree file
* ARGV[1] is a file to write output too

## Unique_phase.pl
Many vairants in WGS data are in perfect linkage disequilibrium with each other. This program finds unique representations from the 'Haploid_geno_phase.csv', which may aid in reducing multiple testing burden in association and/or linkage analysis.
###Additional information
* Output is written to STDOUT so redirect the output to a file
